ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java

```diff
@@ -701,4 +720,254 @@ private static class AccessibleLink {
+        || actionList.contains(AccessibilityNodeInfoCompat.ACTION_LONG_CLICK)
+        || actionList.contains(AccessibilityNodeInfoCompat.ACTION_FOCUS);
+  }
+
+  /**
+   * Returns a cached instance if such is available otherwise a new one.
+   *
+   * @param view The {@link View} to derive the AccessibilityNodeInfo properties from.
+   * @return {@link FlipperObject} containing the properties.
+   */
+  @Nullable
+  public static AccessibilityNodeInfoCompat createNodeInfoFromView(View view) {
+    if (view == null) {
+      return null;
+    }
+
+    final AccessibilityNodeInfoCompat nodeInfo = AccessibilityNodeInfoCompat.obtain();
+
+    // For some unknown reason, Android seems to occasionally throw a NPE from
+    // onInitializeAccessibilityNodeInfo.
+    try {
+      ViewCompat.onInitializeAccessibilityNodeInfo(view, nodeInfo);
+    } catch (NullPointerException e) {
+      if (nodeInfo != null) {
+        nodeInfo.recycle();
+      }
+      return null;
+    }
+
+    return nodeInfo;
+  }
+
+  /**
+   * Creates the text that Google's TalkBack screen reader will read aloud for a given {@link View}.
+   * This may be any combination of the {@link View}'s {@code text}, {@code contentDescription}, and
+   * the {@code text} and {@code contentDescription} of any ancestor {@link View}.
+   *
+   * <p>This description is generally ported over from Google's TalkBack screen reader, and this
+   * should be kept up to date with their implementation (as much as necessary). Details can be seen
+   * in their source code here:
+   *
+   * <p>https://github.com/google/talkback/compositor/src/main/res/raw/compositor.json - search for
+   * "get_description_for_tree", "append_description_for_tree", "description_for_tree_nodes"
+   *
+   * @param view The {@link View} to evaluate.
+   * @param info The default {@link AccessibilityNodeInfoCompat}.
+   * @return {@code String} representing what talkback will say when a {@link View} is focused.
+   */
+  @Nullable
+  public static CharSequence getTalkbackDescription(
+      View view, @Nullable AccessibilityNodeInfoCompat info) {
+    final AccessibilityNodeInfoCompat node =
+        info == null ? createNodeInfoFromView(view) : AccessibilityNodeInfoCompat.obtain(info);
+
+    if (node == null) {
+      return null;
+    }
+    try {
+      final CharSequence contentDescription = node.getContentDescription();
+      final CharSequence nodeText = node.getText();
+
+      final boolean hasNodeText = !TextUtils.isEmpty(nodeText);
+      final boolean isEditText = view instanceof EditText;
+
+      StringBuilder talkbackSegments = new StringBuilder();
+
+      // EditText's prioritize their own text content over a contentDescription so skip this
+      if (!TextUtils.isEmpty(contentDescription) && (!isEditText || !hasNodeText)) {
+        // next add content description
+        talkbackSegments.append(contentDescription + delimiter);
+
+        return removeFinalDelimiter(talkbackSegments);
```

Comment:
Seems unnecessary here to add the delimiter just to immediately remove it.

Reply:
I added and adapted the [logic](https://github.com/facebook/flipper/blob/e44cad5e9912b359378a43ec47c5a49bbca85fc3/android/src/main/java/com/facebook/flipper/plugins/inspector/descriptors/utils/AccessibilityUtil.java#L475-L494) from Flipper method getTalkbackDescription in this [code](https://github.com/facebook/react-native/pull/33690/files#diff-2b24dd2e4bd14c9febf34caed9fe87431885249368148699a00c0c1d23760b76R1001-R1016).

Reply:
@blavalla 

POINT 1)
>Seems unnecessary here to add the delimiter just to immediately remove it.

Thanks. I removed the line.

POINT 2) 
>An element can only really have one role, even if it's made up of the combination of other elements.

https://github.com/facebook/react-native/issues/30719#issuecomment-1143887578

>So there is never a case where both the parent's role and the child's role should be announced. Talkback will always defer to the parent in this case, and whatever role it has, as it only pulls up the contentDescription/text of the child when combining them. This is by design so that you don't end up with elements that have two conflicting roles (for example, "button" and "slider"). Generally speaking, if the child has a role defined, it should probably also be accessible on its own, and therefore won't get pulled up into its parents announcement in the first place.

It is implemented with the following logic in the origin flipper implementation:

https://github.com/facebook/flipper/blob/e44cad5e9912b359378a43ec47c5a49bbca85fc3/android/src/main/java/com/facebook/flipper/plugins/inspector/descriptors/utils/AccessibilityUtil.java#L483-L486
https://github.com/facebook/flipper/blob/e44cad5e9912b359378a43ec47c5a49bbca85fc3/android/src/main/java/com/facebook/flipper/plugins/inspector/descriptors/utils/AccessibilityUtil.java#L508-L511

It is not included in this pull request.